src/base.rs

```diff
@@ -108,6 +119,7 @@ pub struct ObjectTransmissionInformation {
 }
 
 impl ObjectTransmissionInformation {
+    #[cfg(any(feature = "std", feature = "metal"))]
```

Comment:
Since one or the other is required, I think you can skip this. In generally I'd like to see less conditional compilation in this PR. There's quite a lot which is going to make the crate harder to maintain